win32ss/gdi/ntgdi/freetype.c

```diff
@@ -1577,16 +1599,36 @@ IntGdiAddFontResourceEx(PUNICODE_STRING FileName, DWORD Characteristics,
     PVOID Buffer = NULL;
     IO_STATUS_BLOCK Iosb;
     PVOID SectionObject;
-    SIZE_T ViewSize = 0;
+    SIZE_T ViewSize = 0, Length;
     LARGE_INTEGER SectionSize;
     OBJECT_ATTRIBUTES ObjectAttributes;
     GDI_LOAD_FONT LoadFont;
     INT FontCount;
     HANDLE KeyHandle;
+    UNICODE_STRING PathName;
+    WCHAR szPath[MAX_PATH];
```

Comment:
Please do not ever use `MAX_PATH` in the kernel.
The maximum length for paths is `UNICODE_STRING_MAX_CHARS`, and that is way too large of a buffer to go on the stack. So paths should always be paged pool allocations.

Reply:
OK.